src/main/java/networkbook/model/person/PersonSortComparator.java

```diff
@@ -0,0 +1,235 @@
+package networkbook.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static networkbook.commons.util.AppUtil.checkArgument;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+import networkbook.commons.util.ToStringBuilder;
+/**
+ * Provides an ordering for {@code Person} instances based on the values of a specific field.
+ */
+public class PersonSortComparator implements Comparator<Person> {
+
+    /** Comparator that does nothing. */
+    public static final Comparator<Person> EMPTY_COMPARATOR = (Person o1, Person o2) -> 0;
+
+    /** Comparator that puts empty optionals at the end of the list. */
+    public static final Comparator<Optional<?>> OPTIONAL_COMPARATOR = (Optional<?> o1, Optional<?> o2) -> {
+        if (o1.isPresent() && o2.isEmpty()) {
```

Comment:
Can have an assertion here that either `o1` or `o2` must be empty.

Reply:
o1 and o2 can both be present; both present and both empty will return 0 (when both are present/empty, neither will be ordered greater than the other)

Reply:
ok I see